File: db/selfpromo.py
from db import Database

class SelfPromoMsgDB(Database):
    def __init__(self):
        super().__init__()
        self.create()

    def create(self):
        self.execute('''CREATE TABLE IF NOT EXISTS selfpromomsg
                    (msgID INT NOT NULL)''')
        
    def add(self, msg_id):
        self.execute("INSERT INTO selfpromomsg (msgID) VALUES ({})".format(msg_id))
    
    def check(self, msg_id):
        data = self.execute("SELECT * FROM selfpromomsg WHERE msgID={}".format(msg_id))
        return self.check_len(data)

File: db/selfpromo.js
const logger = require('../utility/logger');
const db = require('./database');

class SelfPromoMsgDB {
  constructor() {
    this.create();
  }
  async create() {
    logger.info('Checking/creating selfpromomsg table');
    await db.execute(`CREATE TABLE IF NOT EXISTS selfpromomsg
            (msgID TEXT NOT NULL)`);
  }

  async add(msgID) {
    logger.info(`Adding msg ID #${msgID} to the selfpromomsg table`);
    await db.execute('INSERT INTO selfpromomsg (msgID) VALUES (?)', [msgID]);
  }

  async check(msgID) {
    logger.info(`Checking if msg ID #${msgID} is on the selfpromomsg table`);
    const data = await db.execute('SELECT * FROM selfpromomsg WHERE msgID=?', [msgID]);
    return db.checkLen(data);
  }

}

module.exports = new SelfPromoMsgDB();
